src/iarc7_sensors/landing_detector.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+import sys
+import rospy
+
+from iarc7_msgs.msg import BoolStamped
+from iarc7_msgs.msg import LandingGearContactsStamped
+from geometry_msgs.msg import TwistWithCovarianceStamped
+from nav_msgs.msg import Odometry
+
+from iarc7_safety.SafetyClient import SafetyClient
+
+def switch_callback(msg):
+    global last_switch_message
+    last_switch_message = msg
+
+def odom_callback(msg):
+    global last_height
+    last_height = msg.pose.pose.position.z
+
+last_switch_message = None
+last_height = 0.0
+height_indicates_landed = False
+
+if __name__ == '__main__':
+    rospy.init_node('landing_detector')
+    absolute_velocity_pub = rospy.Publisher('absolute_vel', TwistWithCovarianceStamped, queue_size=0)
+    landing_detected_pub = rospy.Publisher('landing_detected', BoolStamped, queue_size=0)
+    rospy.Subscriber('landing_gear_contact_switches', LandingGearContactsStamped, switch_callback)
+    rospy.Subscriber('odometry/filtered', Odometry, odom_callback)
+
+    rate = rospy.Rate(rospy.get_param('~update_rate'))
+
+    switch_expected_update_rate = rospy.get_param('~switch_expected_update_rate')
+    switch_update_lag_tolerance = rospy.get_param('~switch_update_lag_tolerance')
+    switch_startup_timeout = rospy.get_param('~switch_startup_timeout')
+    landing_detected_height = rospy.get_param('~landing_detected_height')
+    takeoff_detected_height = rospy.get_param('~takeoff_detected_height')
+    takeoff_detected_override_height = rospy.get_param('~takeoff_detected_override_height')
+
+    # Wait for a valid timestamp
+    while rospy.Time.now() == rospy.Time(0) and not rospy.is_shutdown():
+        if rospy.is_shutdown():
```

Comment:
This should get checked after the loop instead of here

Reply:
Fixed